gamemodes/terrortown/gamemode/client/cl_vskin/vgui/dtextentry_ttt2.lua

```diff
@@ -0,0 +1,358 @@
+---
+-- @class PANEL
+-- @section DTextEntryTTT2
+
+local PANEL = {}
+
+local font = "DermaTTT2Text"
+
+---
+-- @accessor number
+-- @realm client
+AccessorFunc(PANEL, "heightMult", "HeightMult")
+
+---
+-- @accessor bool
+-- @realm client
+AccessorFunc(PANEL, "isOnFocus", "IsOnFocus")
+
+---
+-- @accessor string
+-- @realm client
+AccessorFunc(PANEL, "placeholderText", "PlaceholderText")
+
+---
+-- @accessor string
+-- @realm client
+AccessorFunc(PANEL, "curPlaceholderText", "CurrentPlaceholderText")
+
+---
+-- @ignore
+function PANEL:Init()
+	self.TextArea = vgui.Create("DTextEntry", self)
+	self.TextColor = util.GetActiveColor(util.GetChangedColor(util.GetDefaultColor(vskin.GetBackgroundColor()), 25))
+
+	self.TextArea:SetFont(font)
+	self.TextArea:SetTextColor(textColor)
+	self.TextArea:SetCursorColor(textColor)
+
+	self.TextArea.OnValueChange = function(slf,value)
+		-- print("ta.ovc:",slf,value)
+		self:SetValue(value)
+	end
+
+	self.TextArea.OnGetFocus = function(slf)
+		self:SetIsOnFocus(true)
+		self:OnGetFocus()
+	end
+
+	self.TextArea.OnLoseFocus = function(slf)
+		self:SetIsOnFocus(false)
+		self:OnLoseFocus()
+
+		if slf:GetValue() == "" then
+			self:SetCurrentPlaceholderText(self:GetPlaceholderText())
+		else
+			self:SetCurrentPlaceholderText("")
+		end
+	end
+
+	-- This turns off the engine drawing for the text entry
+	self.TextArea:SetPaintBackgroundEnabled(false)
+	self.TextArea:SetPaintBorderEnabled(false)
+	self.TextArea:SetPaintBackground(false)
+
+	-- This turns off the engine drawing of the panel itself
+	self:SetPaintBackgroundEnabled(false)
+	self:SetPaintBorderEnabled(false)
+	self:SetPaintBackground(false)
+
+	-- Sets default values
+	self:SetHeightMult(1)
+	self:SetIsOnFocus(false)
+	self:SetPlaceholderText("searchbar_default_placeholder")
+	self:SetCurrentPlaceholderText("searchbar_default_placeholder")
```

Comment:
Is this necessary? If so please change the placeholder texts to something else to prevent confusion.